File: Random/Swift/Analyzer.swift
func argAnalyze(_ c:inout Customize) -> Bool {

	let args = CommandLine.arguments

	if args.count>1 {
		switch args[1] {
			case "help","-help","--help":
				help()
				return false
			case "version","-version","--version":
				version()
				return false
			default: break
		}
	}

	var sr = ("","")

	enum Responder {
		case none
		case length
		case concurrent
		case rangeFirst
		case rangeSecond
	}
	var r:Responder = .none
	for n in 1..<args.count {
		switch args[n] {
			case "-l","-length": r = .length
			case "-i","-int":
				c.valueType = .int
				r = .rangeFirst
			case "-r","-real":
				c.valueType = .real
				r = .rangeFirst
			case "-parallel","-concurrent":
				r = .concurrent
				fallthrough
			case "-dispatch":
				c.concurrent = .Dispatch
			case "-operation":
				c.concurrent = .Operation
			case "-hidden","-invisible":
				c.visible = false
			default:
			switch r {
				case .length:
					c.length=UInt(args[n]) ?? 1
					r = .none
				case .rangeFirst:
					sr.0 = args[n]
					c.defaultRange = false
					r = .rangeSecond
				case .rangeSecond:
					sr.1 = args[n]
					r = .none
				case .concurrent:
					switch args[n] {
						case "Dispatch": c.concurrent = .Dispatch
						case "Operation": c.concurrent = .Operation
						default: c.concurrent = .Dispatch
					}
					r = .none
				case .none: break
			}
		}
	}

	if sr.0 != "" {
		if c.valueType == .int {
			let ir = (Int(sr.0) ?? 0,Int(sr.1) ?? 0)
			if ir.0==ir.1 {c.intRange = ir.0...ir.0}
			else if ir.0<ir.1 {c.intRange = ir.0...ir.1}
			else if ir.0>ir.1 {c.intRange = ir.1...ir.0}
		}
		if c.valueType == .real {
			let rr = (Double(sr.0) ?? 0,Double(sr.1) ?? 0)
			if rr.0==rr.1 {
				exitByError("範囲指定に問題があります")
			}
			else if rr.0<rr.1 {c.realRange = rr.0..<rr.1}
			else if rr.0>rr.1 {c.realRange = rr.1..<rr.0}
		}
	}

	return true

}

File: Random/C/Analyzer.c
#include <stdlib.h>
#include <string.h>
#include "Structure.h"

void help();
void version();
int cmp(char *,char *);
int emp(char *);

enum Responder {
	None,
	Seed,
	Length,
	RangeFirst,
	RangeSecond
};

int argAnalyze(int argc, char *argv[], Customize *c) {

	enum Responder r=None;
	int n;
	char *firstArg = argv[1];
	char sFirst[30]="";
	char sSecond[30]="";
	char param[30];
	long long int iFirst = 0;
	long long int iSecond = 0;
	long double rFirst = 0;
	long double rSecond = 0;


	if (argc>1) {
		if (cmp(firstArg,"help")||cmp(firstArg,"-help")||cmp(firstArg,"--help")) {
			help();
			return 0;
		}
		if (cmp(firstArg,"version")||cmp(firstArg,"-version")||cmp(firstArg,"--version")) {
			version();
			return 0;
		}
	}


	for (n=1;n<argc;n++) {
		strcpy(param,argv[n]);
		if (cmp(param,"-s")||cmp(param,"-seed")) r=Seed;
		else if (cmp(param,"-i")||cmp(param,"-int")) {
			c->valueType=Int;
			r=RangeFirst;
		}
		else if (cmp(param,"-r")||cmp(param,"-real")) {
			c->valueType=Real;
			r=RangeFirst;
		}
		else if (cmp(param,"-l")||cmp(param,"-length")) r=Length;
		else if (cmp(param,"-hidden")||cmp(param,"-invisible")) c->visible=0;
		else if (r==Seed) {
            if (cmp(param,"none")) c->seedType=Nothing;
            else if (cmp(param,"time")) c->seedType=Time;
            else {
                c->seedType=Custom;
                c->seed=strtouq(param,NULL,10);
            }
			r=None;
		}
		else if (r==Length) {
			c->length=strtouq(param,NULL,10);
			r=None;
		}
		else if (r==RangeFirst) {
			strcpy(sFirst,param);
			c->defaultRange=0;
			r=RangeSecond;
		}
		else if (r==RangeSecond) {
			strcpy(sSecond,param);
			r=None;
		}
	}

	if (!emp(sFirst)) {
		if (c->valueType==Int) {
			iFirst = strtoq(sFirst,NULL,10);
			if (!emp(sSecond)) iSecond = strtoq(sSecond,NULL,10);
			if (iFirst>iSecond) {
				c->max=(long double)iFirst;
				c->min=(long double)iSecond;
			}
			else {
				c->max=(long double)iSecond;
				c->min=(long double)iFirst;
			}
		}
		if (c->valueType==Real) {
			rFirst = strtold(sFirst,NULL);
			if (!emp(sSecond)) rSecond = strtold(sSecond,NULL);
			if (rFirst>rSecond) {
				c->max=rFirst;
				c->min=rSecond;
			}
			else {
				c->max=rSecond;
				c->min=rFirst;
			}
		}
	}

	return 1;

}

int cmp(char *s1,char *s2) {
	return strcmp(s1,s2)==0;
}

int emp(char *s) {
	return strlen(s)==0;
}